File: src/main/java/org/carlosalcina/java/escuela/Profesor.java
package org.carlosalcina.java.escuela;

public class Profesor {
    private String nombre;

    public Profesor(String nombre) {
        this.nombre = nombre;
    }

    public void asignarCurso(Estudiante estudiante, Curso curso) {
        estudiante.asignarCurso(curso);
        System.out.println("Curso " + curso.getNombre() + " asignado a " + estudiante.nombre + " por " + nombre);
    }

    public void asignarCalificacion(Estudiante estudiante, Curso curso, double calificacion) {
        estudiante.asignarCalificacion(curso, calificacion);
        System.out.println("Calificación de " + calificacion + " asignada a " + estudiante.nombre + " en " + curso.getNombre() + " por " + nombre);
    }
}

File: src/main/java/org/carlosalcina/kotlin/escuela/Profesor.kt
package org.carlosalcina.kotlin.escuela

class Profesor(val nombre: String) {
    fun asignarCurso(estudiante: Estudiante, curso: Curso) {
        estudiante.asignarCurso(curso)
        println("Curso ${curso.nombre} asignado a ${estudiante.nombre} por $nombre")
    }

    fun asignarCalificacion(estudiante: Estudiante, curso: Curso, calificacion: Double) {
        estudiante.asignarCalificacion(curso, calificacion)
        println("Calificación de $calificacion asignada a ${estudiante.nombre} en ${curso.nombre} por $nombre")
    }
}